### PR TITLE
feat(uat): implement basic of API communication of paho-python client

### PR DESCRIPTION
**Pull request by eschastlivtsev**
**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-140

**Description of changes:**
- Implement basic setup of Paho python client
- Implement helper scripts
- Implement Basic gRPC API communication functionality

**Why is this change necessary:**
- Part of Python Paho client implementation

**How was this change tested:**
Manual run of mqtt-client-control and python client.
MQTT functionality is not implemented, linking and shutdown steps are successful.

**Test results:**
Python Paho client output
```
python src/main.py test_id
asyncio: [DEBUG] Using selector: SelectSelector
GRPCLib: [INFO] Initialize gRPC library
GRPCLink: [INFO] Making gPRC client connection with 127.0.0.1:47619 as test_id...
GRPCLink: [INFO] Client connection with Control is established, local address is 127.0.0.1
grpc._cython.cygrpc: [DEBUG] Using AsyncIOEngine.POLLER as I/O engine
GRPCLink: [INFO] Handle gRPC requests
GRPCControlServer: [INFO] Server awaiting termination
GRPCControlServer: [INFO] CreateMqttConnection Placeholder TODO
GRPCControlServer: [INFO] shutdownAgent: reason That's it.
GRPCControlServer: [INFO] Server termination done
GRPCLink: [INFO] Shutdown gPRC link
Main: [INFO] Execution done successfully
```
Mqtt Client Control output
```
java -jar target/aws-greengrass-testing-mqtt-client-control.jar
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2023-06-05 18:47:41.333 [main] EngineControlImpl - MQTT client control gRPC server started, listening on 47619
[INFO ] 2023-06-05 18:47:44.631 [grpc-default-executor-0] GRPCDiscoveryServer - RegisterAgent: agentId test_id
[INFO ] 2023-06-05 18:47:44.653 [grpc-default-executor-0] GRPCDiscoveryServer - DiscoveryClient: agentId test_id address 127.0.0.1 port 64173
[INFO ] 2023-06-05 18:47:44.656 [grpc-default-executor-0] EngineControlImpl - Created new agent control for test_id on 127.0.0.1:64173
[INFO ] 2023-06-05 18:47:44.714 [grpc-default-executor-0] ExampleControl - Agent test_id is connected
[INFO ] 2023-06-05 18:47:44.721 [pool-2-thread-1] AgentTestScenario - Playing test scenario for agent id test_id
[ERROR] 2023-06-05 18:47:47.859 [pool-2-thread-1] AgentControlImpl - Couldn't create MQTT connection error '' CONNACK ''
[INFO ] 2023-06-05 18:47:47.866 [pool-2-thread-1] AgentControlImpl - shutdown request sent successfully
[INFO ] 2023-06-05 18:47:47.872 [grpc-default-executor-0] GRPCDiscoveryServer - UnregisterAgent: agentId test_id reason Agent shutdown by OTF request 'That's it.'
[INFO ] 2023-06-05 18:47:47.876 [grpc-default-executor-0] ExampleControl - Agent test_id is disconnected
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bgklika** [comment]
You need to write commit messages in commit lint conventional style please take a look: https://github.com/conventional-changelog/commitlint

To force it you can install husky and commit lint and add pre-commit hook as described in section 'Getting started' on the page linked above.

Currently you can do 
git rebase -i
and rewrite all invalid commit messages.
Message should contains simple form of verb like 'implement'  instead of 'implemented' and so one as an action

File: uat/custom-components/client-python-paho/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+src/dev-env
+*__pycache__*
+scripts/win_setup.bat
```
Comment (bgklika) [inline uat/custom-components/client-python-paho/.gitignore]: That expose names of your local files which is not interested for public repo. You can move all scripts to scripts/local and exclude whole folder if really require these scripts to be local.

File: uat/custom-components/client-python-paho/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+Value 127.0.0.1 will be used by default.
+
+#### Port of control
+The TCP port where gRPC server of Control is listening.
+Value 47619 will be used by default.
+
+### Run
+
+```cmd
+python src/main.py agent1 47619 127.0.0.1
```
Comment (bgklika) [inline uat/custom-components/client-python-paho/README.md]: Replace agent1 to paho-python-agent. 
That will works in both cases but can avoid mistakes

File: uat/custom-components/client-python-paho/src/__init__.py
Comment (bgklika) [inline uat/custom-components/client-python-paho/src/__init__.py:1]: Are these two empty files really required ?

**Review by bgklika — APPROVED** [review]

**Review by bgklika — CHANGES_REQUESTED** [review]
Please rewrite commit messages to meet commit lint requirements.

File: uat/custom-components/client-python-paho/src/__init__.py
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/src/__init__.py:1]: Yes,
__init__.py file is a file that is used to mark a directory as a Python package. Every directory that contains code that you want to be able to import as a simple Python package needs to contain an __init__.py. This includes subdirectories of directories that already have an __init__.py file in them. 

And also, we need __init__ file in the "src" directory for correct pylint work

File: uat/custom-components/client-python-paho/.gitignore (hunk 1 above)
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/.gitignore]: fixed.
Keep "mqtt_client_control_pb2*" as it is output of gRPC generator

File: uat/custom-components/client-python-paho/README.md (hunk 2 above)
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/README.md]: Replaced

**Comment by bgklika** [comment]
Please add
1. License headers to source and script files
2. License check as part of maven build

**Review by bgklika — DISMISSED** [review]

**Comment by eschastlivtsev** [comment]
Added licenses to python files.
License check will be added later

**Review by auarbekov — DISMISSED** [review]

File: .license/style.xml [hunk 3]
```diff
@@ -36,4 +36,14 @@
         <isMultiline>true</isMultiline>
         <padLines>false</padLines>
     </shell>
+    <python>
```
Comment (bgklika) [inline .license/style.xml]: Seems python same as existing shell 
Just remove space from <firstLine># </firstLine> and reuse <shell>

File: uat/custom-components/client-python-paho/pom.xml [hunk 4]
```diff
@@ -78,6 +78,45 @@
...
+                                <exclude>**/mqtt_client_control_pb2*</exclude>
+                                <exclude>**/__pycache__/**</exclude>
+                                <exclude>**/dev-env/**</exclude>
+                            </excludes>
+                        </licenseSet>
+                    </licenseSets>
+                    <mapping>
+                        <py>PYTHON</py>
+                        <bat>BATCH</bat>
+                        <sh>PYTHON</sh>
```
Comment (bgklika) [inline uat/custom-components/client-python-paho/pom.xml]: just use SHELL for both sh and py

**Review by bgklika — APPROVED** [review]

File: uat/custom-components/client-python-paho/scripts/build.sh [hunk 5]
```diff
@@ -1,3 +1,8 @@
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
 #!/bin/sh
```
Comment (bgklika) [inline uat/custom-components/client-python-paho/scripts/build.sh]: Must be first line!

File: uat/custom-components/client-python-paho/scripts/generate_grpc.sh [hunk 6]
```diff
@@ -1,3 +1,8 @@
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
 #!/bin/sh
```
Comment (bgklika) [inline uat/custom-components/client-python-paho/scripts/generate_grpc.sh]: hash bang must be a first line

File: uat/custom-components/client-python-paho/scripts/lint_run.sh [hunk 7]
```diff
@@ -1,3 +1,8 @@
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
 #!/bin/sh
```
Comment (bgklika) [inline uat/custom-components/client-python-paho/scripts/lint_run.sh]: The same

**Review by bgklika — CHANGES_REQUESTED** [review]

File: .license/style.xml (hunk 3 above)
Comment (eschastlivtsev) [reply .license/style.xml]: fixed

File: uat/custom-components/client-python-paho/scripts/build.sh (hunk 5 above)
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/scripts/build.sh]: fixed

File: uat/custom-components/client-python-paho/scripts/lint_run.sh (hunk 7 above)
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/scripts/lint_run.sh]: fixed

File: uat/custom-components/client-python-paho/scripts/generate_grpc.sh (hunk 6 above)
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/scripts/generate_grpc.sh]: fixed

File: uat/custom-components/client-python-paho/pom.xml (hunk 4 above)
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/pom.xml]: fixed

**Review by bgklika — DISMISSED** [review]

**Review by auarbekov — APPROVED** [review]

**Review by auarbekov — DISMISSED** [review]

**Review by bgklika — APPROVED** [review]

**Review by bgklika — DISMISSED** [review]

**Pull request closed without merging**